### PR TITLE
Replace react-redux with the data module in the editor module

### PR DESCRIPTION
**Pull request by youknowriad**
This PR updates all occurrences of react-redux in the `editor` module with `withSelect` and `withDispatch` from the data module. I notice a gain of `100Kb` unminified in the editor module. It's not huge but still a good improvement.

**Testing instructions**
 
Check that:

 - Post visibility, post sticky, post schedule date, post tags, post categories are updated and saved properly
 - Reverting to draft work
 - The code editor works
 - Leaving with dirty changes trigger a modal
 - The writing flow works (arrow navigation)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,6 @@
 		"react-color": "2.13.4",
 		"react-datepicker": "0.61.0",
 		"react-dom": "16.3.0",
-		"react-redux": "5.0.6",
 		"redux": "3.7.2",
```
Comment (gziolo) [inline package.json]: 🎉

**Comment by gziolo** [comment]
Travis complains 😱 😄

File: editor/components/provider/index.js [hunk 2]
```diff
@@ -53,26 +48,18 @@ class EditorProvider extends Component {
...
-
 			// RichText provider:
 			//
 			//  - context.onUndo
 			//  - context.onRedo
 			//  - context.onCreateUndoLevel
 			[
 				RichTextProvider,
-				bindActionCreators( {
+				{
```
Comment (gziolo) [inline editor/components/provider/index.js]: So we have this sorted out, too. Awesome 💯

**Review by gziolo — APPROVED** [review]
Looks great, I tested and didn't discover any regression. Awesome work 🥇 

We still use `redux-multi` and `redux-optimist`. Should we reavelaute their usage?

**Comment by youknowriad** [comment]
> We still use redux-multi and redux-optimist. Should we reavelaute their usage?

`redux-multi` is just small helper allowing array dispatches, it should be fine to remove if we really want to but `redux-optimist` seems important.

**Pull request merged**